upgrade.go

```diff
@@ -212,18 +212,18 @@ func (b *versionedKVBackend) Upgrade(ctx context.Context, s logical.Storage) err
 	}
 
 	writeUpgradeInfoDoneFunc := func(info []byte) {
-	READONLY_LOOP:
 		for {
 			err = s.Put(ctx, &logical.StorageEntry{
 				Key:   path.Join(b.storagePrefix, "upgrading"),
 				Value: info,
 			})
 			switch {
 			case err == nil:
-				break READONLY_LOOP
+				return
 			case err.Error() == logical.ErrSetupReadOnly.Error():
 				time.Sleep(10 * time.Millisecond)
 			default:
+				b.Logger().Error("writing upgrade info resulted in an error", "error", err)
```

Comment:
I think this is a minor issue right?  Like next time a node becomes active it'll try this again, and there aren't any user-visible impacts other than the overhead of traversing the keys a second time to discover they're all already upgraded?  If so, we might want to provide some of that reassurance in the log message.

Reply:
Updated error message with the reassurance